### ISSUE TITLE
ContentFiles to output directory in project which refers a nuget which refers another nuget package

### ISSUE THREAD

**Issue by Bramvanelderen10** [NuGet/Home#9297]
I'm building multiple nuget packages which include content files and can refer each other. These content files should always be copied to output directory.

The problem i'm having is that content files from a nuget package are correctly outputted to the output directory if the project references the nuget directly. But if a nuget package refers another nuget package with contentfiles, these content files will not copy to the output directory and are located in the wrong path.

Does anyone know a solution to this problem?

I've added an example to the repo here: https://github.com/Bramvanelderen10/nugetdemoi

## For context:

So lets say i have the following 2 packages

- Base package
    - resource/contentFileBase.json
- Utilities package
    - resource/contentFileUtilities.json
    - references the base package

In both projects csproj files i have defined the following entries for the content files  

In base:
```
<Content Include="resource/contentFileBase.json">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
      <PackageCopyToOutput>true</PackageCopyToOutput>
</Content>
```  

In utilities:
```
<Content Include="resource/contentFileUtilities.json">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
      <PackageCopyToOutput>true</PackageCopyToOutput>
</Content>
```  

Now i package both the base package and the utilities package using `dotnet pack`. The resulting nuspec for the Utilities package will be generated:

```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
  <metadata>
    <id>Utilities</id>
    <version>0.0.1</version>
    <dependencies>
      <group targetFramework=".NETCoreApp2.2">
        <dependency id="base" version="0.0.1" exclude="Build,Analyzers" />
      </group>
    </dependencies>
    <contentFiles>
      <files include="any/netcoreapp2.2/resources/contentFileUtilities.json" buildAction="Content" copyToOutput="true" />
      <files include="any/netcoreapp2.2/contentFileBase.json" buildAction="Content" />
    </contentFiles>
  </metadata>
</package>
```

As you can immidiatly see in the nuspec the <files> element for contentFileBase.json is invalid. It is missing both the `resources/` path and the `copyToOuput="true"` attribute

Here a screenshot of the setup and the problem:
[![enter image description here][1]][1]

  [1]: https://i.stack.imgur.com/jFmAn.png

Used dotnet cli dotnet pack command. With dotnet core 2.2.4

**Comment by donnie-msft** [comment]
By default, all assets are Private assets. You would need to remove `contentfiles` from the `PrivateAssets` section of your PackageReference.

Add this to your Package Reference

> ` <PackageReference .....>`
>       `<PrivateAssets>analyzers;build</PrivateAssets>`
> ` </PackageReference>`

The documentation for this is https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets

By the way...did you already see this doc and does it make sense? If it's not clear, I'll make sure we have an issue to improve this section of Docs. Thanks for any feedback!

**Comment by Bramvanelderen10** [comment]
@donnie-msft That worked! I read the documentation, where i went wrong was the IncludeAssets element, i thought if i add that with the all value that it would include all files from the package.

I do still have a small problem. Now i do see the file in the resources folder where i expect, but there is a copy still in the root directory. Can this be removed somehow? I've updated the demo project:
![image](https://user-images.githubusercontent.com/6997595/76833607-eef2e680-682b-11ea-9c4a-ffe0e76a7b0b.png)

Thank you very much

**Comment by donnie-msft** [comment]
@Bramvanelderen10 Glad you're making progress, let us know if you have other concerns.

For the small problem, try removing this from base.csproj:
```
  <ItemGroup>
    <None Remove="resources\base.json" />
  </ItemGroup>
```

**Comment by Bramvanelderen10** [comment]
One last question, I tried removing that element but it did not work. I've also tried playing around with different configurations in PrivateAssets, IncludeAssets and ExcludeAssets but all of that did not work.

The only thing that seems to work now is after i use dotnet pack and go to the generated package and edit the nuspec in there. Where i set the buildAction to none manually like:

![image](https://user-images.githubusercontent.com/6997595/77659511-7224e280-6f78-11ea-8572-22efee394bce.png)

But if i set the buildAction to none in visual studio it will change the csproj element to a None element which is not included as contentFile.
